### ISSUE TITLE
KeyError: 'xmpp_jid'

### ISSUE THREAD

**Issue by amckinley** [skoczen/will#55]
This behavior comes and goes between invocations of my bot, but frequently I see the following error whenever will receives a private message. This stops my plugins from executing and I have to restart will to recover. FWIW, we're using a locally-hosted hipchat instance, but there's nothing else weird about our config. Let me know if you'd like additional log output.

```
ERROR    Error processing event handler: <bound method WillBot.message_recieved of <will.main.WillBot object at 0x7f6117bc9bd0>>
Traceback (most recent call last):
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/sleekxmpp/xmlstream/xmlstream.py", line 1693, in _event_runner
    func(*args)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 102, in message_recieved
    self._handle_message_listeners(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 133, in _handle_message_listeners
    msg.room = self.get_room_from_message(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 59, in get_room_from_message
    return self.get_room_by_jid(message.getMucroom())
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 54, in get_room_by_jid
    if room["xmpp_jid"] == jid:
KeyError: 'xmpp_jid'
ERROR    Error handling {jabber:client}message stanza
Traceback (most recent call last):
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/sleekxmpp/xmlstream/xmlstream.py", line 1693, in _event_runner
    func(*args)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 102, in message_recieved
    self._handle_message_listeners(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 133, in _handle_message_listeners
    msg.room = self.get_room_from_message(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 59, in get_room_from_message
    return self.get_room_by_jid(message.getMucroom())
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 54, in get_room_by_jid
    if room["xmpp_jid"] == jid:
KeyError: 'xmpp_jid'
ERROR    'xmpp_jid'
Traceback (most recent call last):
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/sleekxmpp/xmlstream/xmlstream.py", line 1693, in _event_runner
    func(*args)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 102, in message_recieved
    self._handle_message_listeners(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 133, in _handle_message_listeners
    msg.room = self.get_room_from_message(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 59, in get_room_from_message
    return self.get_room_by_jid(message.getMucroom())
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 54, in get_room_by_jid
    if room["xmpp_jid"] == jid:
KeyError: 'xmpp_jid'
ERROR    Error processing event handler: <bound method WillBot.message_recieved of <will.main.WillBot object at 0x7f6117bc9bd0>>
Traceback (most recent call last):
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/sleekxmpp/xmlstream/xmlstream.py", line 1693, in _event_runner
    func(*args)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 102, in message_recieved
    self._handle_message_listeners(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 133, in _handle_message_listeners
    msg.room = self.get_room_from_message(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 59, in get_room_from_message
    return self.get_room_by_jid(message.getMucroom())
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 54, in get_room_by_jid
    if room["xmpp_jid"] == jid:
KeyError: 'xmpp_jid'
ERROR    Error handling {jabber:client}message stanza
Traceback (most recent call last):
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/sleekxmpp/xmlstream/xmlstream.py", line 1693, in _event_runner
    func(*args)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 102, in message_recieved
    self._handle_message_listeners(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 133, in _handle_message_listeners
    msg.room = self.get_room_from_message(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 59, in get_room_from_message
    return self.get_room_by_jid(message.getMucroom())
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 54, in get_room_by_jid
    if room["xmpp_jid"] == jid:
KeyError: 'xmpp_jid'
ERROR    'xmpp_jid'
Traceback (most recent call last):
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/sleekxmpp/xmlstream/xmlstream.py", line 1693, in _event_runner
    func(*args)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 102, in message_recieved
    self._handle_message_listeners(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/listener.py", line 133, in _handle_message_listeners
    msg.room = self.get_room_from_message(msg)
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 59, in get_room_from_message
    return self.get_room_by_jid(message.getMucroom())
  File "/home/austin/.virtualenvs/willbot/local/lib/python2.7/site-packages/will/mixins/room.py", line 54, in get_room_by_jid
    if room["xmpp_jid"] == jid:
KeyError: 'xmpp_jid'
```

**Comment by skoczen** [comment]
As a bump on this, I did see that this has been fixed.  So the V1 token support can be dropped, AFAIK.  Since I'm not blocked on it at present, haven't found time to get it done, but just throwing it out there if someone else can get to it first.

**Comment by skoczen** [comment]
Just a note that this will be fixed in #102.

**Comment by skoczen** [comment]
Closing, as #102 was merged a long time ago!
